pyxem/generators/pdf_generator.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2019 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+
+"""PDF generator and associated tools.
+
+"""
+
+import numpy as np
+
+from hyperspy.signals import Signal1D
+
+from pyxem.signals.reduced_intensity1d import ReducedIntensity1D
+from pyxem.signals.pair_distribution_function1d import PairDistributionFunction1D
+from pyxem.signals import transfer_navigation_axes
+
+
+class PDFGenerator():
+    """Generates a PairDistributionFunction1D signal from a specified
+        ReducedIntensity1D signal.
+
+
+    Parameters
+    ----------
+    signal : ReducedIntensity1D
+        A reduced intensity radial profile.
+    """
+
+    def __init__(self, signal, *args, **kwargs):
+        self.signal = signal
+
+    def get_pdf(self,
+                s_cutoff,
+                r_cutoff=[0, 20],
+                r_increment=0.01
+                ):
+        """ Calculates the pdf from the reduced intensity signal.
+
+        Parameters
+        ----------
+        s_cutoff : list of float
```

Comment:
I don't really like the use of lists for this, can we not just have s_min and s_max etc as kwargs?

Reply:
Changed.